lib/remote/deleteobjecthandler.cpp

```diff
@@ -78,7 +79,29 @@ bool DeleteObjectHandler::HandleRequest(
 
 	bool success = true;
 
+	std::shared_lock wgLock{*waitGroup, std::try_to_lock};
+	if (!wgLock) {
+		HttpUtility::SendJsonError(response, params, 503, "Shutting down.");
+		return true;
+	}
+
 	for (ConfigObject::Ptr obj : objs) {
+		if (!waitGroup->IsLockable()) {
+			if (wgLock) {
+				wgLock.unlock();
+			}
+
+			Dictionary::Ptr fail = new Dictionary({
+				{ "code", 503 },
+				{ "status", "Action skipped: Shutting down."}
+			});
```

Comment:
It would be a good idea to still include "type" and "name" like below so you at least have a chance of knowing which objects weren't deleted:

https://github.com/Icinga/icinga2/blob/fb81054ef39c7cfa27ad00979ed6ede6119ffe60/lib/remote/deleteobjecthandler.cpp#L123-L124

(Same for `ModifyObjectHandler`)